portal-kernel/test/unit/com/liferay/portal/kernel/util/ClassUtilTest.java

```diff
@@ -89,83 +90,83 @@ public void testGetClassesFromAnnotationsWithArrayParameter()
 	}
 
 	@Test
-	public void testGetPathFromURLUnix() throws Exception {
-		testGetPathFromURL(
+	public void testGetPathURIFromURLUnix() throws Exception {
+		testGetPathURIFromURL(
 			"jar:file:/opt/liferay/tomcat/lib/servlet-api.jar" +
 				"!/javax/servlet/Servlet.class",
 			"/opt/liferay/tomcat/lib/servlet-api.jar" +
 				"!/javax/servlet/Servlet.class");
-		testGetPathFromURL(
+		testGetPathURIFromURL(
 			"jar:file:/opt/with%20space/tomcat/lib/servlet-api.jar" +
 				"!/javax/servlet/Servlet.class",
 			"/opt/with space/tomcat/lib/servlet-api.jar" +
 				"!/javax/servlet/Servlet.class");
-		testGetPathFromURL(
+		testGetPathURIFromURL(
 			"file:/opt/liferay/tomcat/classes/javax/servlet/Servlet.class",
 			"/opt/liferay/tomcat/classes/javax/servlet/Servlet.class");
 	}
 
 	@Test
-	public void testGetPathFromURLWeblogic() throws Exception {
-		testGetPathFromURL(
+	public void testGetPathURIFromURLWeblogic() throws Exception {
```

Comment:
@jorgediaz-lr Were these test cases created based on actual tests on these app servers, or reverse engineered from the orignal code?

Reply:
They were created using both approaches.

I analyzed [LPS-15317](https://issues.liferay.com/browse/LPS-15317) that added current code to handle weblogic URLs. The problem is explained in the LPS but the solution was a bit complicated.

But after that, I also installed a Liferay in a weblogic and a wildfly and debugged the code in order to double check the URL format in every call to ClassUtil.